samples/embedded-map/src/components/EmbeddedMap/EmbeddedMapModel.ts

```diff
@@ -77,23 +101,68 @@ export default class EmbeddedMapModel extends ComponentModelBase {
 
         // A new instance is being set - add the event handlers.
         if (instance) {
-            // eslint-disable-next-line @typescript-eslint/no-floating-promises
-            (async () => {
-                // Wait for the initial sync to be setup before listening for
-                // events from Mapillary.
-                await this._syncMaps();
-
-                // Track changes when the user moves to a different node, or when
-                // their bearing/tilt position changes.
-                this.mapillary.on(
-                    Viewer.nodechanged,
-                    this._onPerspectiveChange
-                );
-                this.mapillary.on(Viewer.povchanged, this._onPerspectiveChange);
-            })();
+            // Track changes when the user moves to a different node, or when
+            // their bearing/tilt position changes.
+            this.mapillary.on(Viewer.nodechanged, this._onPerspectiveChange);
+            this.mapillary.on(Viewer.povchanged, this._onPerspectiveChange);
+
+            // Change the mapillary viewer position when the location marker is moved
+            this._viewerUpdateHandle = this.messages.events.locationMarker.updated.subscribe(
+                this._handleViewerUpdate.bind(this)
```

Comment:
You can convert these to arrow functions instead of using `.bind()`.

Reply:
Done